src/lib/components/LicenseField.js

```diff
@@ -0,0 +1,58 @@
+// This file is part of React-Invenio-Deposit
+// Copyright (C) 2020 CERN.
+// Copyright (C) 2020 Northwestern University.
+// Copyright (C) 2020 Cottage Labs LLP.
+//
+// React-Invenio-Deposit is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React, { Component, useMemo } from 'react';
+import PropTypes from 'prop-types';
+
+import { FieldLabel, GroupField, SelectField, TextField } from 'react-invenio-forms';
+import { Field, getIn } from 'formik';
+import { Form } from 'semantic-ui-react';
+
+export function LicenseField({ vocabulary, fieldPath }) {
```

Comment:
(These are more general overall comments before you get too deep)

Since the creation of this issue and now, there were a couple of changes to the [metadata schema](https://github.com/inveniosoftware/docs-invenio-rdm/blob/master/docs/reference/metadata.md#rights-0-n):

- perhaps `RightsField` is a better name now
- There can be multiple licenses selected

We've been using the class-based component approach, so probably best to stick with that (or convince us otherwise to change the rest! :grin:  ).

You can use the `FundingField` as a template for this field. It also had to deal with how the `SelectField` doesn't store what Formik stores in the form. It's a decent example of how to account for that.

Reply:
Thanks for the pointers :-).

I'll rename it to RightsField, and have a look at `FundingField` (I had looked at other fields for inspiration, but not that one).

And I'm happy to follow the class-based approach, and am nowhere near well-versed enough in this to make any convincing arguments ;-).